### DISCUSSION TITLE
Sorting options for local media in media browser

### DISCUSSION THREAD

**Discussion by DIYtechie** [home-assistant/frontend#16190] [Dashboard]
I would really like to have the option to have my local media (video recordings) in the media browser sorted so newest files are shown first per default.

For the time being I use SSH to edit line 189 in /components/media_source/local_source.py to add “, reverse=True”, since there are no sorting options for local media. But this is not ideal since every update overwrites this change of course.

A button in the UI would probably be ideal and a list view would also be great, but just being able to select a sorting order in my config would be a big help.

I believe many others are using the media browser to view images or recordings where the newest is generally the most relevant.

**Comment by nprez83** [comment]
I would also love to have this option, but I'm not good enough to make it happen. 

@karwosts, I noticed you and @bramkragten implemented the addition of a auto/grid/list toggle to the media browser in pr #18256, which I think is a great place to start to get this implemented. 

I also found [this discussion](https://community.home-assistant.io/t/minimum-sorting-on-media-source/460909/4), which points to the file that needs to be modified in order to achieve different sorting options. 

Any chance you could help me add a sort option to the browser, similar to the grid/list toggle you added? I think this would be a great addition for the media browser, especially for those of us who use it to browse through local media such as camera recordings.

**Comment by thefunkygibbon** [comment]
+1 to this.  on top of sorting it would be great to have it being able to change from the default view being thumbnails to the detailed view.

**Comment by Rudd-O** [comment]
+1

I can see in the media browser a list of videos generated by a script that shows me my 3D printer working.  Right now that list is alphabetical, which is 100% useless.

I would like to have a way to browse them by date (most recent first) and, as a stretch goal, the list view should have columns (at least name and date) that are clickable to sort the items on the media folder.  Ideally date created and date last modified would be available columns.  HA already has the ability to present very rich lists — this should be leveraged in the media browser.

I understand not every backend is able to present data about item dates to the frontend.  In those cases a file name will have to suffice.  But the local media source should always be able to query said dates.

**Comment by DIYtechie** [comment]
For some reason my workaround stopped working this year. I even deleted the whole custom component (which was a copy of the original media browser with only reverse:true added). Tried copying it again and change the relevant line (now a higher rownumber / further down), but it no longer works. Such a shame for video recordings.
